### PR TITLE
Refactor for use with new notebook launcher

### PR DESCRIPTION
**Pull request by akchinSTC**
## Related Issues and Dependencies

Marked as noted in Development subsection panel

## This introduces a breaking change

- [ ] Yes
- [ ] No

## This Pull Request implements

Updates image to use latest version of elyra 3.12.0 with the new notebook launcher

## Description

Updates elyra to v3.12.0
Update image structure to latest for new notebook launcher
Update default metadata instance creation

### CODE REVIEW

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thanks for the great PR.

can we not remove other files?
few of them can be used for the automatic build process.

one more thing, In the build with dockerfile or script, i don't see the call to 
`jupyter lab build`
isn't it necessary to show case all the extension that comes along elyra to show up in jupyterlab ?

**Comment by ptitzler** [comment]
> one more thing, In the build with dockerfile or script, i don't see the call to `jupyter lab build` isn't it necessary to show case all the extension that comes along elyra to show up in jupyterlab ?

Not anymore. (Version 3.7 was the last Elyra release that required a rebuild.)

**Comment by harshad16** [comment]
> Not anymore. (Version 3.7 was the last Elyra release that required a rebuild.)

That is amazing :100:

**Comment by akchinSTC** [comment]
```
can we not remove other files?
few of them can be used for the automatic build process.
```
I assume that means we can remove the thoth build stuff? Could you point me to an example repo/template? I believe I just used one of the example repos but was unsure of items to remove. Thanks for the review @harshad16 !

**Comment by harshad16** [comment]
> I assume that means we can remove the thoth build stuff? 

CI `.aicoe-ci.yaml` would enable us to build and push image from this repo.
we can adjust it to dockerfile, it is not bounded by thoth build process.
if there is another plan of build, surely we can remove.
i just thought it would easy build work.

`.thoth.yaml` is used by another bot, which can help with python dependencies updates.
if using requirement.txt or pipfile.


we can remove the .s2i bits, as building using dockerfile.

> Could you point me to an example repo/template? 

don't have a template for this.
i was just suggesting for a point of view of ease of build.

File: .aicoe-ci.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 check:
   - thoth-build
 build:
-  base-image: "quay.io/thoth-station/s2i-minimal-py38-notebook:v0.3.0"
+  base-image: "quay.io/thoth-station/s2i-minimal-py38-notebook:v0.5.0"
   build-stratergy: "Source"
```
Comment (harshad16) [inline .aicoe-ci.yaml:5]:
```suggestion
  build-strategy: Dockerfile
  dockerfile-path: Dockerfile
```

This would allow the CI to build from Dockerfile.

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thanks for the changes.
one suggestion:

File: .aicoe-ci.yaml (hunk 1 above)
Comment (akchinSTC) [reply .aicoe-ci.yaml]: done and thanks

**Review by harshad16 — COMMENTED** [review]
/lgtm

Thanks for all the changes.
as we are removing pipfile and moving to requirements.txt 
please change this value as well in the pr.
https://github.com/opendatahub-io/s2i-lab-elyra/blob/5e215f8ccc29904e257cd8bdc2e82e10a1c1ee84/.thoth.yaml#L3
to `pip`

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

**Comment by harshad16** [comment]
Thank for all the changes
we can merge this and build a tag. :+1:

**Pull request merged**